scripts/cli-build.js

```diff
@@ -0,0 +1,398 @@
+const zlib = require('zlib');
+const { minify } = require('terser');
+const cliTable = require('cli-table');
+const config = require('../config');
+
+let builtWaves = {};
+/**
+ * build a test object
+ * @param {String} configFile, config file path
+ * @param {Object} buildResult, build result
+ * @returns {String} final file contents
+ */
+function buildTest(configFile, buildResult) {
+    let contents,
+        testObject,
+        dirname;
+
+    dirname = path.dirname(configFile);
+    testObject = tokenizePaths(yaml.load(fs.readFileSync(configFile, 'utf8')));
+
+    // skip inactive tests
+    if (testObject.state == 'inactive') {
+        return;
+    }
+
+    // prevent duplicated waves
+    if (builtWaves[testObject.id]) {
+        throw new Error(`Mojito Building - Duplicated waves "${testObject.id}" found in ${builtWaves[testObject.id]} and ${dirname}${path.sep}config.yml`);
+    }
+
+    builtWaves[testObject.id] = `${dirname}${path.sep}config.yml`;
+
+    // remove private data from publishing
+    if (testObject.private) {
+        delete testObject.private;
+    }
+
+    buildResult[testObject.id] = {
+        name: testObject.name,
+        state: testObject.state
+    };
+
+    if (testObject.state == 'live' && testObject.divertTo != null) {
+        buildResult[testObject.id].state = 'divert';
+    }
+
+    contents = JSON.stringify(testObject, null, 4);
+    // inject file contents
+    // shared js and css
+    if (testObject.js) {
+        contents = injectJSFile(testObject.js, contents, dirname);
+    }
+
+    if (testObject.css) {
+        contents = injectCSSFile(testObject.css, contents, dirname);
+    }
+
+    // trigger
+    if (testObject.trigger) {
+        contents = injectJSFile(testObject.trigger, contents, dirname);
+    }
+
+    // recipes
+    var recipes = testObject.recipes || {},
+        recipe;
+    for (var key in recipes) {
+        recipe = recipes[key];
+
+        if (recipe.js) {
+            contents = injectJSFile(recipe.js, contents, dirname);
+        }
+
+        if (recipe.css) {
+            contents = injectCSSFile(recipe.css, contents, dirname);
+        }
+    }
+
+    contents = 'Mojito.addTest(' + contents + ');';
+    buildResult[testObject.id].size = contents.length;
+
+    return contents;
+}
+
+/**
+ * tokenizing the file path, from "1.js" to "{{1.js}}"
+ * @param {Object} testObject 
+ * @returns {Object} tokenized object
+ */
+function tokenizePaths(testObject) {
+    // shared js and css
+    if ('js' in testObject) {
+        if (testObject.js != '') {
+            testObject.js = '{{' + testObject.js + '}}';
+        }
+        else {
+            delete testObject.js;
+        }
+    }
+
+    if ('css' in testObject) {
+        if (testObject.css != '') {
+            testObject.css = '{{' + testObject.css + '}}';
+        }
+        else {
+            delete testObject.css;
+        }
+    }
+
+    // trigger. TODO: throw error if trigger is undefined?
```

Comment:
I think throwing an error here is a good idea. Tests without triggers won't work.

Reply:
Unless you activate them remotely. I'm pretty sure this is possible (it's just not a typical way to activate experiments).

Reply:
@dapperdrop 
A test without trigger can be run by calling test.activate() manually in other tests, let's keep it as is for now?

Reply:
Good points @kingo55 @kingo55 - all good to keep as is for now.